pennylane/transforms/metric_tensor.py

```diff
@@ -292,12 +325,226 @@ def processing_fn(probs):
+        for diffed_op_j, par_idx_j in zip(layer_j.ops, layer_j.param_inds):
+            gen_op_j = _get_gen_op(diffed_op_j, allow_nonunitary, aux_wire)
+            with tape.__class__() as new_tape:
+                # Initialize auxiliary wire
+                qml.Hadamard(wires=aux_wire)
+                # Apply backward cone of first layer
+                for op in layer_i.pre_ops:
+                    qml.apply(op)
+                # Controlled-generator operation of first diff'ed op
+                qml.apply(gen_op_i)
+                # Apply first layer and operations between layers
+                for op in ops_between_cgens:
+                    qml.apply(op)
+                # Controlled-generator operation of second diff'ed op
+                qml.apply(gen_op_j)
+                # Measure X on auxiliary wire
+                qml.expval(qml.PauliX(aux_wire))
+            tapes.append(new_tape)
+            # Memorize to which metric entry this tape belongs
+            ids.append((par_idx_i, par_idx_j))
+
+    return tapes, ids
+
+
+def _metric_tensor_hadamard(tape, allow_nonunitary, aux_wire):
+    r"""Generate the quantum tapes that execute the Hadamard tests
+    to compute the first term of block off-diagonal metric entries
+    and combine them with the covariance matrix-based block diagonal tapes.
+
+    Args:
+        tape (pennylane.tape.QuantumTape): Tape that is being transformed
+        allow_nonunitary (bool): Whether non-unitary operations are allowed
+            in the circuit; passed to ``_get_gen_op``
+        aux_wire (int or pennylane.wires.Wires): Auxiliary wire on which to
+            control the controlled-generator operations. Defaults to
+            ``tape.num_wires`` if not provided
+
+    Returns:
+        list[pennylane.tape.QuantumTape]: Tapes to evaluate the metric tensor
+        callable: processing function to obtain the metric tensor from the tape results
+
+    .. warning::
+
+        This method requires the device used to execute the returned tapes to
+        have an additional wire. This wire may be specified via ``aux_wire``.
+        By default contiguous numbering and an additional wire at the end
+        of the input tape wires are assumed.
+
+    This method is based on computing the first term of the metric tensor with Hadamard
+    tests.
+    This term reads
+
+    .. math ::
+
+        \mathfrak{Re}\left\{\langle \partial_i\psi|\partial_j\psi\rangle\right\}
+
+    and can be computed using an augmented circuit with an additional qubit.
+    See for example `the appendix of this paper <https://arxiv.org/pdf/1804.03023.pdf>`__
+    for details.
+    The block diagonal of the tensor is computed using the covariance matrix approach
+    implemented in :func:`.metric_tensor_cov_matrix`. In addition, we may extract the
+    factors for the second terms :math:`\langle \psi|\partial_j\psi\rangle`
+    of the block off-diagonal tensor from the tape results for the covariance matrix.

```

Comment:
Because the covariance matrix uses expectation values <\psi | \partial_i \psi> and this is the vector which is put into an outer product with itself to get the second term of the full metric tensor :) This actually means we save some circuits, but the scaling of the full metric tensor remains quadratic.